File: kirun-js/src/engine/util/ArrayUtil.ts
export class ArrayUtil {
    public static removeAListFrom(source: any[], removeList: any[]): void {
        if (!removeList || !source || !source.length || !removeList.length) return;

        const e: Set<any> = new Set<any>(removeList);

        for (let i = 0; i < source.length; i++) {
            if (!e.has(source[i])) continue;
            source.splice(i, 1);
            i--;
        }
    }

    public static of<K>(...k: K[]): K[] {
        const copy: K[] = new Array(k.length);

        for (let i = 0; i < k.length; i++) copy[i] = k[i];

        return copy;
    }

    private constructor() {}
}

File: kirun-py/src/kirun_py/util/array_util.py
from typing import Any


class ArrayUtil:

    @staticmethod
    def remove_a_list_from(source: list, remove_list: list) -> None:
        if not remove_list or not source:
            return
        to_remove = set()
        for item in remove_list:
            try:
                to_remove.add(item)
            except TypeError:
                pass

        i = 0
        while i < len(source):
            try:
                if source[i] in to_remove:
                    source.pop(i)
                    continue
            except TypeError:
                pass
            i += 1

    @staticmethod
    def of(*items: Any) -> list:
        return list(items)
